x/participation/keeper/available_allocations_test.go

```diff
@@ -0,0 +1,76 @@
+package keeper_test
+
+import (
+	"strconv"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/participation/types"
+)
+
+func TestAvailableAllocationsGet(t *testing.T) {
+	sdkCtx, tk, _ := testkeeper.NewTestSetup(t)
+
+	invalidAddress := strconv.Itoa(1)
+	allocationPrice := types.AllocationPrice{Bonded: sdk.NewInt(100)}
+
+	tk.ParticipationKeeper.SetParams(sdkCtx, types.Params{
+		AllocationPrice: allocationPrice,
+	})
+
+	validAddress := sample.Address()
+	validAddressNoUse := sample.Address()
+	addressNegativeDelegations := sample.Address()
+
+	tk.DelegateN(sdkCtx, validAddress, 100, 10)
+	tk.DelegateN(sdkCtx, validAddressNoUse, 100, 10)
+	tk.DelegateN(sdkCtx, addressNegativeDelegations, -100, 10)
+
+	tk.ParticipationKeeper.SetUsedAllocations(sdkCtx, types.UsedAllocations{
+		Address:        validAddress,
+		NumAllocations: 2,
+	})
+
+	for _, tc := range []struct {
+		desc       string
+		address    string
+		allocation uint64
+		wantError  bool
+	}{
+		{
```

Comment:
Based on the comment above, we should have a test case where `totalAllocation < usedAllocation` and it returns 0